content/docs/101-why-effect/index.md

```diff
@@ -0,0 +1,151 @@
+---
+title: Why should I use this?
+---
+
+Effect offers a different approach writing TypeScript than what most people are used to. However, to buy into using Effect, you don't have to use the entire system. You can use pieces of the system independently, and migrate your codebase incrementally without committing to everything at once.
+<br />
+With that said, the rest of this page is dedicated to trying to demonstrate to you that Effect is worth buying into as an entire system for writing TypeScript, not just as a small set of utility functions in random parts of your program.
+<br />
+Let's jump into an example.
+<br />
+For the following set of examples, observe how the structure of the code changes, and don't worry about how we implement a specific action like sending an email.
+<br />
+
+## Simple program (no error handling)
+
+Let's imagine a simple TypeScript program that creates a user in the database and invites a user via email.
+
+```ts twoslash
+declare function createUserInDB(args: { name: string; email: string }): { id: string; name: string; email: string }
+
+declare function sendUserInvite(email: string): void
+
+const createUser = async (name: string, email: string) => {
+  const newUser = await createUserInDB({ name, email })
+  await sendUserInvite(email)
+  return newUser
+}
+```
+
+Everything looks great right? We created a user in the database, and then sent an email.
+<br />
+However, what do we do when these functions have errors? Often, it's easy to pretend that they won't happen, but they will eventually.
+<br />
+
+## Simple program (basic error handling)
+
+```ts twoslash
+declare function createUserInDB(args: { name: string; email: string }): { id: string; name: string; email: string }
+
+declare function sendUserInvite(email: string): void
+
+const createUser = async (name: string, email: string) => {
+  try {
+    const newUser = await createUserInDB({ name, email })
+    await sendUserInvite(email)
+    return newUser
+  } catch (e) {
+    return e
+  }
+}
+```
+
+Now, we've done a basic form of "error handling" by wrapping everything in a `try/catch`. This is the easy way to do it, and it's surprising how often code is left like this. The only difference between this and the previous example is that we prevented our app from crashing, but we didn't actually do anything to handle the errors that each function can throw. When our entire codebase looks like this, it becomes really hard to make sense of what's happening in our codebase when things go wrong.
+<br />
+
+There's also some additional problems here:
+
+// TODO these docs don't handle numerical lists correctly
+
+    1. We have no idea what kinds of errors could be thrown by `createUserInDB` or `sendUserInvite`. I just have to assume that they will throw exceptions and always wrap my code in a `try/catch`.
```

Comment:
Disregard this if you do this later on, but it might help drive the point home if you explicitly point out that TypeScript does natively provide a way to express that a function might throw -- which effectively means you cannot reason about how your program will behave at runtime from the signatures alone.

That makes any program's behavior impossible to predict without "breaking open" each and every function and reading the entire implementation.

Which seems to be what you're getting at here -- in the past when I've discussed this subject this teammates, the fact that TypeScript does not have an answer for this (and that nobody talks about it) does seem to pique people's interest

Reply:
I agree, I've had the same epiphany when talking to teammates about this subject. If I end up tweaking this or including this in the docs in some way, will definitely make sure to mention this.